source/extensions/filters/common/expr/context.h

```diff
@@ -89,6 +89,8 @@ constexpr absl::string_view RouteName = "route_name";
 constexpr absl::string_view RouteMetadata = "route_metadata";
 constexpr absl::string_view UpstreamHostMetadata = "upstream_host_metadata";
 constexpr absl::string_view FilterChainName = "filter_chain_name";
+constexpr absl::string_view ListenerMetadata = "listener_metadata";
```

Comment:
In what consideration not to use `listener.metadata` and `listener.direction`?

Reply:
You mean Wasm attributes? We should recommend the standard `xds` attributes once we finish replacing them, and then deprecate those one-off cases.

Reply:
My bad, I misunderstood.